cli/index.ts

```diff
@@ -202,131 +223,131 @@ createYargsCli({
       options: [
         fullRefreshOption,
         nodesOption,
         includeDepsOption,
         defaultSchemaOption,
         assertionSchemaOption,
-        credentialsOption,
-        {
-          name: "result-path",
-          option: {
-            describe: "Optional path where executed graph JSON should be written.",
-            type: "string"
-          }
-        }
+        credentialsOption
       ],
-      processFn: argv => {
-        console.log("Project status: starting...");
-
-        const projectConfig = require(path.resolve(argv["project-config"], "dataform.json"));
-        const credentials = utils.readCredentials(projectConfig.warehouse, argv["credentials"]);
-
-        compile({
+      processFn: async argv => {
+        const compiledGraph = await compile({
           projectDir: argv["project-dir"],
           defaultSchemaOverride: argv["default-schema"],
           assertionSchemaOverride: argv["assertion-schema"]
-        })
-          .then(graph => {
-            console.log("Project status: build...");
-
-            return build(
-              graph,
-              {
-                fullRefresh: argv["full-refresh"],
-                nodes: argv["nodes"],
-                includeDependencies: argv["include-deps"]
-              },
-              credentials
-            );
-          })
-          .then(graph => {
-            const tasksAmount = graph.nodes.reduce((prev, item) => prev + item.tasks.length, 0);
-            console.log(
-              `Project status: ready for run ${
-                graph.nodes.length
-              } node(s) with ${tasksAmount} task(s)`
-            );
-            console.log("Project status: running...");
+        });
+        if (compiledGraph.graphErrors) {
+          logGraphErrors(compiledGraph.graphErrors);
+          return;
+        }
+        const credentials = utils.readCredentials(
+          compiledGraph.projectConfig.warehouse,
+          argv["credentials"]
+        );
+        const executionGraph = await build(
+          compiledGraph,
+          {
+            fullRefresh: argv["full-refresh"],
+            nodes: argv["nodes"],
+            includeDependencies: argv["include-deps"]
+          },
+          credentials
+        );
 
-            const runner = run(graph, credentials);
-            process.on("SIGINT", () => {
-              runner.cancel();
+        const runner = run(executionGraph, credentials);
+        process.on("SIGINT", () => {
+          runner.cancel();
+        });
+        const executedGraph = await runner.resultPromise();
+        const prettyPrintedExecutedGraph = prettyJsonStringify(executedGraph);
+        console.log(prettyPrintedExecutedGraph);
+        if (!executedGraph.ok) {
+          executedGraph.nodes
+            .filter(node => node.status === dataform.NodeExecutionStatus.FAILED)
+            .forEach(node => {
+              console.log(errorOutputFmtString, `Execution failed on node "${node.name}":`);
```

Comment:
Do we have a similar error message for skipped and disabled nodes?

Reply:
No - my opinion is that for skipped or disabled nodes, those were explicitly set by the user, so it makes no sense to error out in those cases.